src/modules/roc_netio/target_libuv/roc_netio/udp_sender_port.cpp

```diff
@@ -111,14 +111,24 @@ bool UdpSenderPort::open() {
     return true;
 }
 
-void UdpSenderPort::async_close() {
+bool UdpSenderPort::async_close() {
     core::Mutex::Lock lock(mutex_);
 
     stopped_ = true;
 
+    if (!handle_initialized_ && !write_sem_initialized_) {
```

Comment:
NITPICK: we have the following in several places:

```cpp
 if (!handle_initialized_ && !write_sem_initialized_) 
```

Probably we could inline it in: ```was_opened``` ?

Reply:
I've extracted `fully_closed_()`.